File: backend/src/main/java/com/app/backend/responses/cauhoi/CauHoiResponse.java
package com.app.backend.responses.cauhoi;

import com.app.backend.models.CauHoi;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CauHoiResponse {
    private Long id;

    @JsonProperty("noi_dung")
    private String noiDung;

    @JsonProperty("do_kho")
    private String doKho;

    @JsonProperty("loai_noi_dung")
    private String loaiNoiDung;

    @JsonProperty("duong_dan_tep")
    private String duongDanTep;

    @JsonProperty("lua_chon_a")
    private String luaChonA;

    @JsonProperty("lua_chon_b")
    private String luaChonB;

    @JsonProperty("lua_chon_c")
    private String luaChonC;

    @JsonProperty("lua_chon_d")
    private String luaChonD;

    @JsonProperty("dap_an_dung")
    private Character dapAnDung;

    @JsonProperty("giai_thich")
    private String giaiThich;

    @JsonProperty("bo_cau_hoi_id")
    private Long boCauHoiId;

    @JsonProperty("bo_cau_hoi_tieu_de")
    private String boCauHoiTieuDe;

    public static CauHoiResponse from(CauHoi entity) {
        return CauHoiResponse.builder()
                .id(entity.getId())
                .noiDung(entity.getNoiDung())
                .doKho(entity.getDoKho())
                .loaiNoiDung(entity.getLoaiNoiDung())
                .duongDanTep(entity.getDuongDanTep())
                .luaChonA(entity.getLuaChonA())
                .luaChonB(entity.getLuaChonB())
                .luaChonC(entity.getLuaChonC())
                .luaChonD(entity.getLuaChonD())
                .dapAnDung(entity.getDapAnDung())
                .giaiThich(entity.getGiaiThich())
                .boCauHoiId(entity.getBoCauHoi() != null ? entity.getBoCauHoi().getId() : null)
                .boCauHoiTieuDe(entity.getBoCauHoi() != null ? entity.getBoCauHoi().getTieuDe() : null)
                .build();
    }
}

File: frontend/src/app/responses/cauhoi/cauhoi-response.ts
export interface CauHoiResponse {
  id: number;
  noi_dung: string;
  do_kho: 'DE' | 'TRUNG_BINH' | 'KHO';
  loai_noi_dung: 'VAN_BAN' | 'HINH_ANH' | 'AM_THANH' | 'VIDEO';
  duong_dan_tep?: string | null;

  lua_chon_a: string;
  lua_chon_b: string;
  lua_chon_c: string;
  lua_chon_d: string;
  dap_an_dung: 'A' | 'B' | 'C' | 'D';
  giai_thich?: string;

  bo_cau_hoi_id: number;
  bo_cau_hoi_tieu_de?: string;
}
